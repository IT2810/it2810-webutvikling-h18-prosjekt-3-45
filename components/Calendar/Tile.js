import React, { Component } from 'react';
import { View, StyleSheet } from 'react-native';
import { Text } from 'native-base';
import format from 'date-fns/format';

class Tile extends Component {
  render() {
    const doneCount = this.props.todos.filter(todo => todo.done).length;

    return (
      <View
        style={[styles.tile, this.props.index % 7 === 0 && styles.leftTile]}
      >
        <Text style={styles.day}>{format(this.props.day, 'D')}</Text>

        <Text style={styles.badge}>
          {doneCount}/{this.props.todos.length}
        </Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  tile: {
    width: `${100 / 7}%`,
    padding: 2,
    height: 50,
    position: 'relative',
    borderLeftWidth: 1,
    borderBottomWidth: 1,
    borderColor: '#EEE',
  },

  leftTile: {
    borderLeftWidth: 0,
  },

  day: {
    fontWeight: 'bold',
  },

  badge: {
    position: 'absolute',
    right: 2,
    bottom: 2,
  },
});

export default Tile;
