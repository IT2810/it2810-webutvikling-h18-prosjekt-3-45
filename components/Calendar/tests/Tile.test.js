import React from 'react';
import renderer from 'react-test-renderer';
import { Tile } from '../Tile';

jest.mock('../../TodoAdder', () => () => 'TodoAdder');

describe('Tile', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Tile todos={[]} day={new Date('Sun, 07 Oct 2018 16:03:43 GMT')} />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
