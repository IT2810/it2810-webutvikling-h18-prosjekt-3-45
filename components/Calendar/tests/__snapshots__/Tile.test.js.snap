// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#EEE",
        "borderLeftWidth": 1,
        "height": 50,
        "padding": 2,
        "position": "relative",
        "width": "14.285714285714286%",
      },
      false,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        },
        Object {
          "fontWeight": "bold",
        },
      ]
    }
    uppercase={false}
  >
    7
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        },
        Object {
          "bottom": 2,
          "position": "absolute",
          "right": 2,
        },
      ]
    }
    uppercase={false}
  >
    0
    /
    0
  </Text>
</View>
`;
